Dockerfile

```diff
@@ -5,7 +5,15 @@ RUN chown -R 1000:1000 /opt/conda/
 
 # install headers to build regex as part of Black
 # https://github.com/psf/black/issues/1112
-RUN apt update && apt install -y build-essential python3-dev
+# and also install required components for
+# warp-transducer build (make, cmake gcc-6)
+RUN apt-get update && apt-get install \
+    build-essential  \
+    python3-dev cmake make \
+    software-properties-common -y && \
+    echo "deb  http://deb.debian.org/debian  stretch main" >> /etc/apt/sources.list && \
+    apt-get update && \
+    apt-get install gcc-6 g++-6 -y
```

Comment:
I've sorted the imports. Waiting on the CI build with gcc-6 install removed.

Reply:
I also get build-essential installing gcc-6 when I build the docker locally. However the CI fails with the gcc-6 install removed as it instead installs gcc-8 for some reason. Output from the CI as follows:
```
The following additional packages will be installed:
  binutils binutils-common binutils-x86-64-linux-gnu cmake-data cpp cpp-8 dbus
  dh-python dirmngr distro-info-data dmsetup dpkg-dev fakeroot g++ g++-8 gcc
  gcc-8 gir1.2-glib-2.0 gir1.2-packagekitglib-1.0 gnupg gnupg-l10n gnupg-utils
  gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm iso-codes
  libalgorithm-diff-perl libalgorithm-diff-xs-perl libalgorithm-merge-perl
  libapparmor1 libappstream4 libapt-inst2.0 libarchive13 libargon2-1 libasan5
  libassuan0 libatomic1 libbinutils libc-dev-bin libc6-dev libcap2 libcap2-bin
  libcc1-0 libcryptsetup12 libcurl4 libdbus-1-3 libdevmapper1.02.1
  libdpkg-perl libelf1 libexpat1-dev libfakeroot libfile-fcntllock-perl
  libgcc-8-dev libgirepository-1.0-1 libglib2.0-0 libglib2.0-bin libgomp1
  libgpm2 libgstreamer1.0-0 libidn11 libip4tc0 libisl19 libitm1 libjson-c3
  libjsoncpp1 libkmod2 libksba8 liblocale-gettext-perl liblsan0 libmpc3
  libmpdec2 libmpfr6 libmpx2 libncurses6 libncursesw6 libnpth0 libnss-systemd
  libpackagekit-glib2-18 libpam-cap libpam-systemd libpolkit-agent-1-0
  libpolkit-backend-1-0 libpolkit-gobject-1-0 libprocps7 libpython3-dev
  libpython3-stdlib libpython3.7 libpython3.7-dev libpython3.7-minimal
  libpython3.7-stdlib libquadmath0 librhash0 libstdc++-8-dev libstemmer0d
  libsystemd0 libtinfo6 libtsan0 libubsan1 libuv1 libyaml-0-2 linux-libc-dev
  lsb-release manpages manpages-dev packagekit packagekit-tools
  pinentry-curses policykit-1 procps psmisc python-apt-common python3
  python3-apt python3-dbus python3-distro-info python3-distutils python3-gi
  python3-lib2to3 python3-minimal python3-pycurl python3-software-properties
  python3.7 python3.7-dev python3.7-minimal systemd systemd-sysv
  unattended-upgrades
```
Re-adding the gcc-6 install.

Reply:
I've realised during this investigation that `make` is installed with `build-essential` so removing it from here

Reply:
Nice! This is so much cleaner! Thanks for the explanation as to why this was occurring - I now have the new miniconda image locally and am getting the expected behaviour.

Reply:
update: I've removed `software-properties-common` as this was necessary to add the stretch repository. 
The black pre-commit issue just required `gcc` installation and the warp-transducer build requires `make` so I've replaced `build-essential` -> `make` to avoid the `gcc-8` install.

Reply:
update2: we can also remove the `python3-dev` because the black error has been fixed upstream. 
This now reads: 

```
RUN apt-get update && apt-get install -y \
    cmake \
    g++-7 \
    gcc-7 \
    make
```